ring/lifecycler.go

```diff
@@ -415,6 +415,11 @@ func (i *Lifecycler) setReadOnlyState(readOnly bool, readOnlyLastUpdated time.Ti
 	defer i.stateMtx.Unlock()
 	i.readOnly = readOnly
 	i.readOnlyLastUpdated = readOnlyLastUpdated
+	if readOnly {
+		i.lifecyclerMetrics.readonly.Set(1)
+	} else {
+		i.lifecyclerMetrics.readonly.Set(0)
```

Comment:
What if rather than exporting a `0` value (which will be the case the majority of the time, and always for lifecyclers not using read-only state) we just stop exporting the metric entirely when not in read-only state?

Reply:
You are right that this will be mostly 0, but I don't think removing the metric when it's 0 is a good practice. It makes it difficult to see the difference between metric being 0 vs metric not working correctly for some reason. It's also easier to work with metric that exists all the time vs metric that can be absent.